IPython/paths.py

```diff
@@ -72,7 +72,7 @@ def get_ipython_dir() -> str:
                     " using a temp directory.".format(parent))
             ipdir = tempfile.mkdtemp()
         else:
-            os.makedirs(ipdir)
+            ensure_dir_exists(ipdir)
```

Comment:
`ensure_dir_exists` sets different permissions by default, also, `os.makedirs` has `exist_ok` parameter which is what needed, no?

Reply:
on double-checking, os.makedirs uses 511 mode by default; ensure_dir_exists uses 755. Those seem significantly different and I don't know what the downstream consequences are. Can you advise?

Reply:
I am neither a maintainer nor the author of this code, just had a question that I could not get answer by looking at the code/commit message.